src/datastore/postgres-store.ts

```diff
@@ -5618,6 +5618,36 @@ export class PgDataStore
         `,
         [address, maxBlockHeight]
       );
+
+      if (query.rowCount > 0) {
+        query.rows = query.rows.filter(async row => {
```

Comment:
I don't think this is working as intended. The `filter` function does not handle promises, so it will see the `truthy` promise and rows will never be filtered, and these promises will still be evaluating even after the function exists since nothing is awaiting them.